examples/camera-controller/webrtc_manager/WebRTCProviderClient.cpp

```diff
@@ -169,6 +235,15 @@ void WebRTCProviderClient::OnDeviceConnectionFailure(void * context, const Scope
     self->OnDone(nullptr);
 }
 
+void WebRTCProviderClient::OnSessionEstablishTimeout(chip::System::Layer * systemLayer, void * appState)
+{
+    WebRTCProviderClient * self = reinterpret_cast<WebRTCProviderClient *>(appState);
+    VerifyOrReturn(self != nullptr, ChipLogError(Camera, "OnSessionEstablishTimeout: context is null"));
+
+    ChipLogError(Camera, "WebRTC Session establishment has timed out!");
+    self->MoveToState(State::Idle);
```

Comment:
This should also remove the WebRTCSession that was created after receiving the ProvideOffer response, right?